vm/context.ts

```diff
@@ -466,8 +490,10 @@ module J2ME {
     }
 
     start(frames: Frame[]) {
-      frames.unshift(Frame.Start);
-      this.frames = frames;
+      this.frames = [Frame.Start];
+      for (var i = 0; i < frames.length; i++) {
+        this.pushFrame(frames[i]);
+      }
```

Comment:
I like this refactoring too, since it avoids side-effects in the caller. It does result in the allocation of an extra array, but this method seems unlikely to be hot, so I don't think it matters.

Nevertheless, note that *this.frames* is already initialized to a new empty array in the *Context* constructor, so you could initialize it to `[Frame.Start]` there and then merely push the *frames* onto it here.